File: app/src/main/java/com/github/aldychris/mobiusexample/CounterEffectHandler.java
package com.github.aldychris.mobiusexample;

import com.spotify.mobius.Connection;
import com.spotify.mobius.functions.Consumer;


class CounterEffectHandler {
    static Connection<CounterEffect> effectHandler(Consumer<CounterEvent> eventConsumer) {
        return new Connection<CounterEffect>() {
            @Override
            public void accept(CounterEffect effect) {
                if (effect == CounterEffect.REPORT_ERROR_NEGATIVE) {
                    System.out.println("error!");
                }
            }

            @Override
            public void dispose() {
                // No resources to release, so we can leave this empty.
            }
        };
    }
}

File: app/src/main/java/com/github/aldychris/mobiusexample/CounterEffectHandler.kt
package com.github.aldychris.mobiusexample

import com.spotify.mobius.Connection
import com.spotify.mobius.functions.Consumer
import io.reactivex.Observable

internal object CounterEffectHandler {
    @JvmStatic
    fun effectHandler(eventConsumer: Consumer<CounterEvent?>?): Connection<CounterEffect> {
        return object : Connection<CounterEffect> {
            override fun accept(effect: CounterEffect) {
                if (effect === IsError) {
                    println("error!")
                }
            }

            override fun dispose() {
                // No resources to release, so we can leave this empty.
            }
        }
    }

    @JvmStatic
    fun rxEffectHandler(effects: Observable<CounterEffect>): Observable<CounterEvent> {
        return effects.ofType(IsError.javaClass)
            .flatMap {
                Observable.just(InputError(it))
            }
    }
}